Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -555,7 +556,7 @@ extension NSManagedObjectContext: MessageDatabaseSession {
             throw ClientError.ChannelDoesNotExist(cid: cid)
         }
 
-        let message = MessageDTO.loadOrCreate(id: .newUniqueId, context: self, cache: nil)
+        let message = MessageDTO.loadOrCreate(id: messageId, context: self, cache: nil)
```

Comment:
Nit: if we make `messageId: MessageId?` optional here, the `messageId ?? .newUniqueId` logic is only applied here instead of duplicating it whenever we use the Updaters 👍